cartographer_ros/cartographer_ros/node.cc

```diff
@@ -441,14 +448,30 @@ void Node::StartTrajectoryWithDefaultTopics(const TrajectoryOptions& options) {
   AddTrajectory(options, DefaultSensorTopics());
 }
 
-std::unordered_set<std::string> Node::ComputeDefaultTopics(
-    const TrajectoryOptions& options) {
-  carto::common::MutexLocker lock(&mutex_);
-  return ComputeExpectedTopics(options, DefaultSensorTopics());
+std::vector<
+    std::set<cartographer::mapping::TrajectoryBuilderInterface::SensorId>>
+Node::ComputeDefaultSensorIdsForMultipleTrajectories(
```

Comment:
I recommend changing this to `ForMultipleBags`, because the order of the bags may not be the same as the order of trajectories (line 231).

Reply:
Also, if you load a .pbstream, the trajectories may not begin with zero (you actually don't know how many serialized trajectories are inside). I think that's another reason to make this namespacing per-bag.

Reply:
Done. Right, indexes are not trajectory_id, but bag_index.